src/chat/index.ts

```diff
@@ -382,6 +383,10 @@ export class OpenRouterChatLanguageModel implements LanguageModelV2 {
                 response.usage?.cost_details?.upstream_inference_cost ?? 0,
             },
           },
+        } as {
+          provider: string;
+          reasoning_details: ReasoningDetailUnion[];
+          usage: OpenRouterUsageAccounting;
```

Comment:
We should add this type into the provider metadata schema instead, I think it's satisfied against a type